Do not use default sorting when panel initializes it.

dmolineus opened the pull request:
The sorting panel did not have any affects if a default sorting is defined. It would be added as additional sortings. This pull request starts the panel initialization with an empty sorting definition and will fallback to the defualt one.

discordier commented:
I guess this code could be moved into a helper method in the `ViewHelpers` class like this:
```php
public function setSortingFromPanel($panel, $dataConfig)
{
        // Store default sorting start initializing the panel with an empty sorting.
        $sorting = $dataConfig->getSorting();
        $dataConfig->setSorting(array());
        $panel->initialize($dataConfig);
         // Restore default sorting if panel did not set any.
        if ($sorting && !$dataConfig->getSorting()) {
            $dataConfig->setSorting($sorting);
        }
}
```
Otherwise we are introducing even more duplicated code.

dmolineus commented:
Moved it into the helper method. Actually the creation of the missing sorting are duplicate code as well. So I moved it all into an `initializeSorting` method.
